components/create_comment/create_comment.jsx

```diff
@@ -344,16 +347,24 @@ export default class CreateComment extends React.PureComponent {
             return;
         }
 
-        const {draft} = this.state;
+        const {draft, target} = this.state;
 
         let newMessage = '';
         if (draft.message === '') {
             newMessage = `:${emojiAlias}: `;
-        } else if ((/\s+$/).test(draft.message)) {
-            // Check whether there is already a blank at the end of the current message
-            newMessage = `${draft.message}:${emojiAlias}: `;
         } else {
-            newMessage = `${draft.message} :${emojiAlias}: `;
+            const {draft: {message}} = this.state;
+            const {firstPiece, lastPiece} = splitMessageBasedOnCaretPosition.call(this, message);
+
+            // check whether the first piece of the message is empty when cursor is placed at beginning of message and avoid adding an empty string at the beginning of the message
+            newMessage = firstPiece === '' ? `:${emojiAlias}: ${lastPiece} ` : `${firstPiece} :${emojiAlias}: ${lastPiece} `;
+
+            const newCaretPosition = `${firstPiece} :${emojiAlias}: `.length;
+            repositionCaret(target, newCaretPosition);
```

Comment:
Instead of using a helper function that runs with `requestAnimationFrame`, maybe it would be cleaner to use the second argument of `setState` to update the actual caret position after the state has been updated?
```javascript
this.setState({
    caretPosition: newCaretPosition,
}, () => {
    Utils.setCaretPosition(textbox, newCaretPosition);
});
```

Reply:
@hmhealey I have made the suggested changes on this. Works fine, thanks! :)